frontend/src/App.tsx

```diff
@@ -49,63 +49,144 @@ import '@ionic/react/css/display.css'
 
 Amplify.configure(awsExports)
 
-const App: React.FC = () => (
-  <IonApp>
-    <IonReactRouter>
-      <IonTabs>
-        <IonRouterOutlet>
-          {/* Moments routes  */}
-          <Route exact path={'/moments'} component={MomentsListView} />
-          <Route path={`/moments/details/:id`} component={MomentsDetailView} />
-          <Route path={`/moments/create`} component={MomentsCreateNewView} />
-
-          {/* Reflections routes */}
-          <Route exact path={'/reflections'} component={ReflectionsListView} />
-          <Route
-            path={`/reflections/details/:id`}
-            component={ReflectionsDetailView}
-          />
-          <Route
-            path={`/reflections/create`}
-            component={ReflectionsCreateNewView}
-          />
-
-          {/* profile routes */}
-          <Route exact path={'/profile'} component={ProfileDetailView} />
-          <Route
-            path={`/profile/changepassword`}
-            component={ProfileChangePasswordView}
-          />
-
-          {/* progress routes */}
-          <Route exact path={'/progress'} component={ProgressDetailView} />
-        </IonRouterOutlet>
-        <IonTabBar slot="bottom">
-          <IonTabButton tab="moments" href="/moments">
-            <IonIcon icon={image} />
-            <IonLabel>Momente</IonLabel>
-          </IonTabButton>
-          <IonTabButton tab="reflections" href="/reflections">
-            <IonIcon icon={albums} />
-            <IonLabel>Reflexionen</IonLabel>
-          </IonTabButton>
-          <IonTabButton tab="profile" href="/profile">
-            <IonIcon icon={person} />
-            <IonLabel>Profil</IonLabel>
-          </IonTabButton>
-          <IonTabButton tab="progress" href="/progress">
-            <IonIcon icon={barChart} />
-            <IonLabel>Fortschritt</IonLabel>
-          </IonTabButton>
-        </IonTabBar>
-      </IonTabs>
-    </IonReactRouter>
-  </IonApp>
-)
-
-// export default App
+const App: React.FC = () => {
+  const [authState, setAuthState] = useState(AuthState.LoggedOut)
+
+  // Details for all auth states can be found here:
+  // https://docs.amplify.aws/lib/auth/auth-events/q/platform/js
+  const authListener = (data: any) => {
+    switch (data.payload.event) {
+      case 'signIn':
+      case 'signUp':
+        Auth.currentAuthenticatedUser().then((user: any) =>
+          setAuthState(user ? AuthState.LoggedIn : AuthState.ConfirmSignUp),
+        )
+        break
+      case 'signOut':
+        setAuthState(AuthState.LoggedOut)
+    }
+  }
+
+  useEffect(() => {
+    // initial check if authenticated or not
+    Auth.currentAuthenticatedUser()
+      .then((user: any) =>
+        setAuthState(user ? AuthState.LoggedIn : AuthState.ConfirmSignUp),
+      )
+      .catch(() => setAuthState(AuthState.LoggedOut))
+
+    // continuously checking for auth events
+    Hub.listen('auth', authListener)
+
+    return () => {
+      Hub.remove('auth', authListener)
+    }
+  }, [])
```

Comment:
I think that this mechanism is kind of terrible?
By using the switch statement I have more control over what to show and handling information to the respective components.

If we use the HOC with custom components they are basically rendered above each other and we have to check in each component what the `authState` is.

With my solution we have that abstracted away in the `App.tsx` and each component can focus on its own functionality.

I thought that's a more elegant solution but curious what you think.

Reply:
I also would rather use the `HoC` since it handles the routing when someone is not logged in and the less we need to manage the better. I agree that it isn't perfect that amplify still "renders" them. 
But in the end, has @DaemonLoki built the "same" than amplify offers out-of-the-box that's why I would go with the HoC, but I don't have that strong opinion on it. 

I am totally okay going with the custom implementation, but then it has to work the same way and be tested, e.g. what will happen if my token is expired and I open the app, will it use the refresh-token, to gather a new one, etc.